R/pkg/R/SQLContext.R

```diff
@@ -189,19 +238,67 @@ createDataFrame <- function(data, schema = NULL, samplingRatio = 1.0,
           x
         }
       }
+      data[] <- lapply(data, cleanCols)
 
-      # drop factors and wrap lists
-      data <- setNames(lapply(data, cleanCols), NULL)
+      args <- list(FUN = list, SIMPLIFY = FALSE, USE.NAMES = FALSE)
+      if (arrowEnabled) {
+        shouldUseArrow <- tryCatch({
+          stopifnot(length(data) > 0)
+          dataHead <- head(data, 1)
+          # Currenty Arrow optimization does not support POSIXct and raw for now.
+          # Also, it does not support explicit float type set by users. It leads to
+          # incorrect conversion. We will fall back to the path without Arrow optimization.
+          if (any(sapply(dataHead, function(x) is(x, "POSIXct")))) {
+            stop("Arrow optimization with R DataFrame does not support POSIXct type yet.")
+          }
+          if (any(sapply(dataHead, is.raw))) {
+            stop("Arrow optimization with R DataFrame does not support raw type yet.")
+          }
+          if (inherits(schema, "structType")) {
+            if (any(sapply(schema$fields(), function(x) x$dataType.toString() == "FloatType"))) {
+              stop("Arrow optimization with R DataFrame does not support FloatType type yet.")
+            }
+          }
+          firstRow <- do.call(mapply, append(args, dataHead))[[1]]
+          fileName <- writeToTempFileInArrow(data, numPartitions)
+          tryCatch(
+            jrddInArrow <- callJStatic("org.apache.spark.sql.api.r.SQLUtils",
+                                       "readArrowStreamFromFile",
+                                       sparkSession,
+                                       fileName),
+          finally = {
+            file.remove(fileName)
```

Comment:
unlink?

Reply:
I believe either way is fine.

Reply:
yes, just more consistent. I also don't know for sure why all other instances are calling unlink